packages/flame/lib/src/components/mixins/has_game_ref.dart

```diff
@@ -28,4 +30,13 @@ mixin HasGameRef<T extends FlameGame> on Component {
     super.onRemove();
     _gameRef = null;
   }
+
+  /// Directly assigns (and override if one is already set) a [gameRef] to the
+  /// component.
+  ///
+  /// This is meant to be used only for testing purposes.
+  @visibleForTesting
+  void mockGameRef(T gameRef) {
```

Comment:
Maybe this could be a simple setter?

Reply:
I would like to keep prefix `mock` to make a clear statement that this is a mock. it could me a setter, but the name I wouldn't like to change.

Reply:
So, like, the name reinforces the idea that this is for test purposes only? I guess it makes sense then.

Reply:
Yep, that is my idea.